key/trie/trie.go

```diff
@@ -0,0 +1,323 @@
+// Package trie provides an implementation of a XOR Trie
+package trie
+
+import (
+	"errors"
+
+	"github.com/plprobelab/go-kademlia/key"
+)
+
+var ErrMismatchedKeyLength = errors.New("key length does not match existing keys")
```

Comment:
rm in favor of the `key` package error

Reply:
I think it belongs here. The key package error is for invalid keys. Here all the keys are valid keys, but one does not match the format of the others